common/src/main/java/com/wynntils/features/chat/ChatMentionFeature.java

```diff
@@ -61,69 +61,42 @@ private Pattern buildPattern() {
     public void onChat(ChatMessageReceivedEvent e) {
         Component message = e.getMessage();
 
-        Matcher looseMatcher = mentionPattern.matcher(ComponentUtils.getUnformatted(message));
+        StyledText styledText = StyledText.fromComponent(message);
 
-        if (looseMatcher.find()) {
-            if (markMention.get()) {
-                e.setMessage(rewriteComponentTree(message));
-            }
-            if (dingMention.get()) {
-                McUtils.playSoundUI(SoundEvents.NOTE_BLOCK_PLING.value());
-            }
-        }
-    }
-
-    private Component rewriteComponentTree(Component comp) {
-        // Make a copy of the component without the siblings
-        MutableComponent curr = MutableComponent.create(comp.getContents()).withStyle(comp.getStyle());
-        // .getString() is used here as it gives formattingchars when those exist. It is needed for guild messages
-        // because wynn still uses legacy coloring for it.
-        String text = curr.getString();
-
-        // if current component has text check it for mentions and rewrite the component if necessary
-        if (!text.isEmpty()) {
-            curr = rewriteMentions(curr, text, comp.getStyle());
-        }
+        StyledTextPart partFinding = styledText.getPartFinding(mentionPattern, PartStyle.StyleType.NONE);
 
-        // process and append siblings
-        for (Component c : comp.getSiblings()) {
-            curr.append(rewriteComponentTree(c));
+        // If the part is null, it means that the mention is in two different parts, we likely don't want to mark it.
```

Comment:
.. or it means the chat message did not mention the user..? :)

Reply:
Yes :). This code was shuffled around quite a bit, the comment was left a bit outdated.